src/community/modularity.jl

```diff
@@ -0,0 +1,18 @@
+"""
+Computes Newman's modularity `Q`
+for graph `g` given the partitioning `c`.
+"""
+function modularity(g::Graph, c)
+    Q = 0.
+    m = 2 * ne(g)
+    m == 0 && return 0.
+    for u in vertices(g)
+        for v in vertices(g)
+            c[u] != c[v] && continue
+            a = has_edge(g, u, v) ? 1 : 0
+            Q += a - degree(g, u)*degree(g, v) / m
+        end
+    end
+    Q = abs(Q) < 1e-10 ? 0  : Q
```

Comment:
Because for complete graphs Q=0, but due to float addition non-associativity I was obtaing Q~10e- 17, therefore I chose 1e-10 as an arbitrary but reasonable cut-off. Anyway I'll avoid the problem turning to integers sums.